crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -294,18 +294,84 @@ impl Mesh {
         self.attributes.insert(name.into(), values);
     }
 
-    /// Retrieve the data currently set behind a vertex attribute.
+    /// Retrieves an immutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute(&self, name: impl Into<Cow<'static, str>>) -> Option<&VertexAttributeValues> {
         self.attributes.get(&name.into())
     }
 
+    /// Retrieves a mutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute_mut(
         &mut self,
         name: impl Into<Cow<'static, str>>,
     ) -> Option<&mut VertexAttributeValues> {
         self.attributes.get_mut(&name.into())
     }
 
+    /// Retrieves multiple mutable references to _distinct_ vertex attributes.
+    /// Returns None at index i if the mesh had no data for the vertex attribute given at index i.
+    ///
+    /// # Panics
+    /// Panics if any of the given attribute names resolve to the same attribute, as this would otherwise allow aliased mutability.
+    ///
+    /// # Example usage
+    /// ```
+    /// use crate::bevy_render::mesh::shape;
+    /// use crate::bevy_render::mesh::Mesh;
+    /// let mut mesh = Mesh::from(shape::Box::default());
+    /// let [positions, normals] = mesh.attribute_multi_mut([Mesh::ATTRIBUTE_POSITION, Mesh::ATTRIBUTE_NORMAL]);
+    /// ```
+    pub fn attribute_multi_mut<const N: usize>(
+        &mut self,
+        names: [impl Into<Cow<'static, str>>; N],
+    ) -> [Option<&mut VertexAttributeValues>; N] {
+        // SAFE: We verify that all the references returned are unique,
+        // therefore no aliased mutability is possible.
+        unsafe {
+            const INIT_POINTER: Option<*mut VertexAttributeValues> = None;
+            let mut pointers = [INIT_POINTER; N];
+
+            for (pointer, name) in pointers.iter_mut().zip(names) {
+                // Get raw mutable pointers to the attributes.
+                // Note that these pointers may indeed alias if the user gave duplicate names!
+                *pointer = self.attributes.get_mut(&name.into()).map(|a| a as *mut _);
+            }
+
+            // Verify that there are no duplicate pointers.
+            // This is critical to ensure no aliased mutability!
+            for i in 1..pointers.len() {
```

Comment:
This approach to duplicate checking will be `O(n^2)`. Can we use `Hash` to get an `O(n)` solution here instead? Do we even care? My intuition suggests that n will be very small here.

Reply:
We could use a temporary HashSet here, yes but as you alude to, I suspect the heap allocation of that set would be more expensive than the O(n^2) here, where n is probably very rarely more than 5.

Reply:
Added a comment to make this explicit.